internal/remoteproc/remoteproc.go

```diff
@@ -17,10 +18,27 @@ const (
 	rprocFirmwareFileName     = "firmware"
 )
 
-var (
-	rprocFirmwareStorePath = rootpath.Join("lib", "firmware")
-	rprocClassPath         = rootpath.Join("sys", "class", "remoteproc")
-)
+var rprocClassPath = rootpath.Join("sys", "class", "remoteproc")
+
+func setFirmwareStorePath() (string, error) {
+	// Check if kernel has custom firmware path configured
+	systemFirmwarePath := rootpath.Join("lib", "firmware")
+	if customPath, err := os.ReadFile("/sys/module/firmware_class/parameters/path"); err == nil {
+		if path := strings.TrimSpace(string(customPath)); path != "" {
+			systemFirmwarePath = path
+		}
+	}
+	// Try system firmware path first, fall back to user home
+	rprocFirmwareStorePath := systemFirmwarePath
+	if _, err := os.Stat(rprocFirmwareStorePath); err != nil {
+		currentUser, err := user.Current()
+		if err != nil {
+			return "", fmt.Errorf("failed to get current user: %w", err)
+		}
+		rprocFirmwareStorePath = filepath.Join(currentUser.HomeDir, "firmware")
+	}
+	return rprocFirmwareStorePath, nil
```

Comment:
Comments = need for functions?

```suggestion
	firmwareStorePath := getSystemFirmwareStorePath()
	if _, err := os.State(firmwareStorePath); err == nil {
	    return firmwareStorePath, nil
	}
    usersFirmwarePath, err := getUsersFirmwareStorePath()
    if err != nil {
       return "", fmt.Errorf("can't determine safe space to store firmware") // <- it'd be nice to provide more detail to the user (what we tried), but also it might be too much? I suspect this error will rarely happen.
    }
    return usersFirmwarePath, nil
```